File: test_archives/py_tests/batch1/test_function8.py
from functions.function8 import function8
import pytest
def test_function8_1():
    assert function8('Hello ', 10, [1, 2], True) == 'Hello 1210'

def test_function8_2():
    assert function8('Hi ', 15, [4, 5], False) == '4515'

File: test_archives/cpp_tests/batch1/test_function8.cpp
#include "../catch.hpp"
#include "../functions/function8.cpp"

TEST_CASE( "Testing function8 with the first test case" ) {
    REQUIRE( function8("Hello ", 10, {1, 2}, true) == "Hello 1210" );
}

TEST_CASE( "Testing function8 with the second test case" ) {
    REQUIRE( function8("Hi ", 15, {4, 5}, false) == "4515" );
}